docs: update local mode guide to use `0.13` terms (#4106)
garden-io/garden: docs/guides/running-service-in-local-mode.md

## Glossary

* **Local app** - a locally deployed and running application which is supposed to replace a `Deploy` action configured
  in _local mode_.

> * **Target k8s workload** -
>   a [Kubernetes Workload](https://kubernetes.io/docs/concepts/workloads/) produced from a `Deploy` action configured in
>   _local mode_ and deployed in a target k8s cluster.

Reviewer: ```suggestion
* **Target k8s workload** -
  a [Kubernetes Workload](https://kubernetes.io/docs/concepts/workloads/) produced from a `Deploy` action with a 
  _local mode_ configuration that's deployed to the target kubernetes cluster.
```

## Introduction

> By configuring a `Deploy` action in _local mode_, one can "replace" a target k8s workload in a k8s cluster with a
> local app.

Reviewer: ```suggestion
A local application can be used instead of a kubernetes workload by adding a _local mode_ configuration 
to the `Deploy action`
```

I don't like the k8s abbreviation, we don't use it elsewhere and it sounds too slangy for my liking.
Also I think this reads a bit clearer.

### Number of the services in local mode

Only one container can be run in local mode for each [`kubernetes`](../reference/action-types/Deploy/kubernetes.md)

> or [`helm`](../reference/action-types/Deploy/helm.md) service. This limitation is planned to be removed in Garden

Author: Right, it should be "`Deploy` action".

Also, those checks create some extra traffic to the local service. That might be noisy and unnecessary if the local
service is running in the debugger.

### Configuring local mode for `container` action type

```yaml
kind: Build
name: node-service

> type: container

Reviewer: I don't think the build is needed here

---

kind: Deploy
name: node-service
type: container
build: node-service
...
spec:
  localMode:
    ports:
      - local: 8090 # The port of the local app, will be used for port-forward setup.
        remote: 8080 # The port of the remote app, will be used for port-forward setup.
    # Starts the local app which will replace the target one in the k8s cluster.
    # Optional. If not specified, then the local app should be started manually.
    command: [ npm, run, serve ]
    # Defines how to restart the local app on failure/exit.
    # Optional. If not specified, then the default values will be applied.
    restart:
      delayMsec: 2000 # 2 sec delay between local app restarts
      max: 100 # limit restart attempts to 100
    ...
...
```

An example can be found in the [`local-mode project`](../../examples/local-mode).

### Configuring local mode for `kubernetes` and `helm` action types

```yaml
kind: Build
name: backend

> type: container

Reviewer: I don't think the build is needed here
